src/libs/auth.js

```diff
@@ -0,0 +1,98 @@
+import _ from 'lodash/fp'
+import { Sam, Rawls, Leo } from 'src/libs/ajax'
+import * as Config from 'src/libs/config'
+import * as Utils from 'src/libs/utils'
+
+export const authStore = Utils.atom({
+  isSignedIn: undefined,
+  isRegistered: undefined
+})
+
+export const initializeAuth = _.memoize(async () => {
+  await new Promise(resolve => window.gapi.load('auth2', resolve))
+  await window.gapi.auth2.init({ clientId: await Config.getGoogleClientId() })
+  authStore.update(state => ({ ...state, isSignedIn: getAuthInstance().isSignedIn.get() }))
+  getAuthInstance().isSignedIn.listen(isSignedIn => {
+    return authStore.update(state => {
+      return { ...state, isSignedIn, isRegistered: isSignedIn ? state.isRegistered : undefined }
+    })
+  })
+})
+
+authStore.subscribe((state, oldState) => {
+  if (!oldState.isSignedIn && state.isSignedIn) {
+    Sam.getUserStatus().then(response => {
+      if (response.status === 404) {
+        return false
+      } else if (!response.ok) {
+        throw response
+      } else {
+        return response.json().then(({ enabled: { ldap } }) => !!ldap)
+      }
+    }).then(isRegistered => {
+      authStore.update(state => ({ ...state, isRegistered }))
+    }) // report failure
```

Comment:
Is this a TODO?

Reply:
Yes. To be fixed shortly by the error message overhaul.

Reply:
Can you put i.e. TODO or Need to report failure or Don't currently report failure so that's clear?